util/threadpool_imp.cc

```diff
@@ -100,6 +103,35 @@ struct ThreadPoolImpl::Impl {
   // Set the thread priority.
   void SetThreadPriority(Env::Priority priority) { priority_ = priority; }
 
+  int ReserveThreads(int threads_to_be_reserved) {
+    std::unique_lock<std::mutex> lock(mu_);
+    int reserved_threads_in_success = threads_to_be_reserved;
+    // We can reserve at most num_waiting_ in total so the number of threads
+    // that can be reserved might be fewer than the desired one
+    if (reserved_threads_ + threads_to_be_reserved > num_waiting_) {
+      reserved_threads_in_success = num_waiting_ - reserved_threads_;
+      reserved_threads_ = num_waiting_;
+    } else {
+      reserved_threads_ += threads_to_be_reserved;
+    }
```

Comment:
Is it equivalent to 
```
reserved_threads_in_success = std::min(num_waiting_ - reserved_threads_, threads_to_be_reserved)
reserved_threads_ += reserved_threads_in_success;
```

Similar case for `ReleaseThreads()`

Reply:
Good catch! Will apply this change soon.